### DISCUSSION TITLE
Cannot find a C preprocessor when configure the PETSc

### DISCUSSION THREAD

**Discussion by 1Northward1** [mdolab/dafoam#353] [Installation]
Hi,

I am installing the DAFoam on Ubuntu 20.04 and exactly followed the command in the installation tutorial. 
There was a problem configuring PETSc:

===============================================================================
             Configuring PETSc to compile on your system                       
===============================================================================
TESTING: checkCPreprocessor from config.setCompilers(config/BuildSystem/config/setCompilers.py:699)                                                                                                                                                  *******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
-------------------------------------------------------------------------------
Cannot find a C preprocessor
*******************************************************************************

With the help of my friends, I activated the conda environment on the new terminal and then compiled it again. This problem can be solved, but an error occurred when compiling the DAFoam:

/bin/sh: /home/Mars/dafoam/packages/miniconda3/lib/libtinfo.so.6: no version information available (required by /bin/sh)
In file included from /home/Mars/dafoam/packages/petsc-3.14.6/include/petscis.h:7,
                 from /home/Mars/dafoam/packages/petsc-3.14.6/include/petscvec.h:9,
                 from /home/Mars/dafoam/packages/petsc-3.14.6/include/petscmat.h:6,
                 from /home/Mars/dafoam/packages/petsc-3.14.6/include/petscpc.h:6,
                 from /home/Mars/dafoam/packages/petsc-3.14.6/include/petscksp.h:6,
                 from DAUtility/DAUtility.H:16,
                 from DAUtility/DAUtility.C:8:
/home/Mars/dafoam/packages/petsc-3.14.6/include/petscsys.h:219:6: error: #error "PETSc was configured with one OpenMPI mpi.h version but now appears to be compiling using a different OpenMPI mpi.h version"
  219 | #    error "PETSc was configured with one OpenMPI mpi.h version but now appears to be compiling using a different OpenMPI mpi.h version"
      |      ^~~~~

What can be the possible cause? 
I would appreciate it if you help me.

Best regards,
Mars

**Comment by friedenhe** [comment]
`PETSc was configured with one OpenMPI mpi.h version but now appears to be compiling using a different OpenMPI mpi.h version` The error says that you use two different MPI versions to compile Petsc and DAFoam. Double check if you have followed the exact steps to compile DAFoam.
